### PR TITLE
JAMES-1713 Graph should throw when loop detected

### PR DESCRIPTION
**Pull request by rouazana**

### CODE REVIEW

**Comment by aduprat** [comment]
:+1:

File: server/protocols/jmap/src/test/java/org/apache/james/jmap/utils/DependencyGraphTest.java [hunk 1]
```diff
@@ -114,11 +115,47 @@ public void getBuildChainOnComplexTreeShouldWork() {
...
+    
+    @Test(expected=CycleDetectedException.class)
+    public void getBuildChainOnTreeWithLoopShouldFail() {
+        //Given
+        Commit a = new Commit("A");
+        Commit b = new Commit("B", a);
+        a.setParent(b);
+        List<Commit> input = ImmutableList.of(a, b);
+        DependencyGraph<Commit> testee = new DependencyGraph<>(Commit::getParent);
+        input.stream().forEach(testee::registerItem);
```
Comment (fvn-linagora) [inline server/protocols/jmap/src/test/java/org/apache/james/jmap/utils/DependencyGraphTest.java]: as input is only used once, we could inline here (Stream.of(...).forEach

File: server/protocols/jmap/src/test/java/org/apache/james/jmap/utils/DependencyGraphTest.java [hunk 2]
```diff
@@ -114,11 +115,47 @@ public void getBuildChainOnComplexTreeShouldWork() {
...
+        Commit b = new Commit("B", a);
+        
+        Commit c = new Commit("C");
+        Commit d = new Commit("D", c);
+        Commit e = new Commit("E", d);
+        Commit f = new Commit("F", e);
+        c.setParent(f);
+        List<Commit> input = ImmutableList.of(a, b, c, d, e, f);
+        DependencyGraph<Commit> testee = new DependencyGraph<>(Commit::getParent);
+        input.stream().forEach(testee::registerItem);
```
Comment (fvn-linagora) [inline server/protocols/jmap/src/test/java/org/apache/james/jmap/utils/DependencyGraphTest.java]: input building could be inlined here, as well

Comment (rouazana) [inline server/protocols/jmap/src/test/java/org/apache/james/jmap/utils/DependencyGraphTest.java]: ok

File: server/protocols/jmap/src/test/java/org/apache/james/jmap/utils/DependencyGraphTest.java (hunk 1 above)
Comment (rouazana) [inline server/protocols/jmap/src/test/java/org/apache/james/jmap/utils/DependencyGraphTest.java]: ok

**Comment by fvn-linagora** [comment]
:+1:

**Comment by aduprat** [comment]
:+1:

**Pull request merged**